panther_localization/launch/ekf.launch.py

```diff
@@ -50,27 +62,60 @@ def generate_launch_description():
         choices=["True", "False"],
     )
 
-    robot_localization_node = Node(
+    ekf_local = Node(
         package="robot_localization",
         executable="ekf_node",
-        name="ekf_node",
+        name="ekf_local",
         parameters=[ekf_config_path, {"tf_prefix": namespace}],
         namespace=namespace,
         remappings=[
             ("/diagnostics", "diagnostics"),
-            ("enable", "ekf_node/enable"),
-            ("set_pose", "ekf_node/set_pose"),
-            ("toggle", "ekf_node/toggle"),
+            ("enable", "~/enable"),
+            ("set_pose", "~/set_pose"),
+            ("toggle", "~/toggle"),
+            ("odometry/filtered", "odometry/filtered/local"),
+        ],
+    )
+
+    ekf_global = Node(
+        package="robot_localization",
+        executable="ekf_node",
+        name="ekf_global",
+        parameters=[ekf_config_path, {"tf_prefix": namespace}],
+        namespace=namespace,
+        remappings=[
+            ("/diagnostics", "diagnostics"),
+            ("enable", "~/enable"),
+            ("set_pose", "~/set_pose"),
+            ("toggle", "~/toggle"),
+            ("odometry/filtered", "odometry/filtered/global"),
+        ],
+        condition=LaunchConfigurationEquals("ekf_configuration", "global"),
+    )
+
+    navsat_transform = Node(
+        package="robot_localization",
+        executable="navsat_transform_node",
+        name="navsat_transform",
+        parameters=[ekf_config_path, {"tf_prefix": namespace}],
+        namespace=namespace,
+        remappings=[
+            ("gps/fix", "gps/fix"),
+            ("odometry/filtered", "odometry/filtered/global"),
+            ("odometry/gps", "_odometry/gps"),
         ],
-        emulate_tty=True,
+        condition=LaunchConfigurationEquals("ekf_configuration", "global"),
     )
```

Comment:
Do we need a 3x node definition? Underneath these are 2 different executables. Wouldn't be better to define one `ekf_node` with two configs or one config for two configurations?

Reply:
This is how it is usually done:
![image](https://github.com/husarion/panther_ros/assets/126687345/aa0bc23f-b268-4f5f-838f-2bd6d56a65af)

There are some alternatives:
1. Single `ekf_node` with arg `ekf_mode:=global/local`. In such case the `/odometry/filtered` will contain:
- relative position to start point  for `local` arg (x:=5 means move 5m stright).
- adds + imu orientation + gps to calculate ENU position for `global` (x:=5 means move 5m to the east).

2. Two nodes `ekf_local`, `ekf_global` (as it is) with arg `ekf_mode:=global/local`.  In such case:
- setting arg to `local` creates `/odometry/filtered/local` which contain relative position to the init pose
- setting arg to `global` add new `ekf_global` node which uses `/odometry/filtered/local` + imu orientation + gps to calculate ENU position

3. Two nodes `ekf_relative`, `ekf_enu` and two type of arg `gps_data:=true/false` `ekf_mode:=relative/enu/both`. 
This will most likely result in you having 4 ekf configurations and switching them accordingly:
- `ekf_relative_with_gps.yaml`
- `ekf_relative.yaml`
- `ekf_enu_with_gps.yaml`
- `ekf_enu.yaml`

Reply:
Ok, now I got you more precise, and the proposed launch makes more sense 👍🏼 
I wonder if approach no.3 isn't an overhead?

One thing that came to my mind you can consider is leaving the config file as it is now and defining two separate launch files:
1. `ekf.launch.py` - local EKF
2. `gps_localization.launch.py`/`global_ekf.launch.py` (or any appropriate name) - local + global + navsat

Seems to me more intuitive from a user perspective but demands more work maintaining the files. The decision is yours here.